staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -64,13 +66,28 @@ func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.Gr
 			statusSchemaValidator: statusSchemaValidator,
 		},
 		structuralSchemas: structuralSchemas,
+		kind:              kind,
 	}
 }
 
 func (a customResourceStrategy) NamespaceScoped() bool {
 	return a.namespaceScoped
 }
 
+// GetResetFields returns the set of fields that get reset by the strategy
+// and should not be modified by the user.
+func (a customResourceStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{}
+
+	if a.status != nil {
```

Comment:
Is a.status going to be set prior to calling this method?

Reply:
Yes, it looks like it gets created here https://github.com/kubernetes/kubernetes/pull/99661/files#diff-01e7a2059fe62e95ec32e2589c01b7218100de327647c641c334d38f2c5aa06eL737-R752

on the condition that subresources.Status is not nil.

I believe this is the only place `customresource.NewStrategy` is called.